### PR TITLE
[Bin] Handle possible PHPStan Stubs runtime locations

### PR DESCRIPTION
**Pull request by samsonasik**
Point possible vendor locations by use the` __DIR__` as start to locate
Like in https://github.com/rectorphp/rector/pull/5581 that may not detected in https://getrector.org/ which uses docker to run

### CODE REVIEW

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

File: bin/rector [hunk 1]
```diff
@@ -1,14 +1,43 @@
...
-    require_once 'phar://vendor/phpstan/phpstan/phpstan.phar/stubs/runtime/ReflectionUnionType.php';
-    require_once 'phar://vendor/phpstan/phpstan/phpstan.phar/stubs/runtime/Attribute.php';
-}
+/**
+ * @see https://github.com/phpstan/phpstan/issues/4541#issuecomment-779434916
+ *
+ * Point possible vendor locations by use the __DIR__ as start to locate
+ * @see https://github.com/rectorphp/rector/pull/5581 that may not detected in https://getrector.org/ which uses docker to run
+ */
+(static function () {
```
Comment (TomasVotruba) [inline bin/rector:10]: This is look like very magically. Class with method would be better.

The `bin/rector` should be clear of any PHP code. It was my mistake to put phar includes here in the first place. Better locatio would be probably the `ContainerFactory`, so any container created is valid without any extra includes.

**Comment by TomasVotruba** [comment]
What issue is this trying to solve?

**Comment by samsonasik** [comment]
When `rector` bin run from non-current project directory, like via docker: 

https://github.com/rectorphp/getrector.org/blob/814a94687e29db7566fef491a120c77744c1ad43/bin/run-demo.sh#L32-L41

Currently, the `getrector.org` is using PHP 8, so everything is fine, however, when it run in PHP < 8, it will got error undefined class `ReflectionUnionType` as not exists like in the following PR https://github.com/rectorphp/rector/pull/5569

The next patch, which uses `file_exists()` : https://github.com/rectorphp/rector/pull/5581 will only patch for run rector in current directory, for non current directory, it will completely skipped, and if it run on PHP < 8, the error will appear again.

**Comment by TomasVotruba** [comment]
I see. Let's add this autoload then.

Could you move it to ContainerFactory as I write above?

**Comment by samsonasik** [comment]
I can't found `ContainerFactory` class, is it the `RectorContainerFactory`, or should we apply to another `StubLoader` class, eg: `Rector\Core\Stubs\PHPStanStubLoader` ?

**Comment by samsonasik** [comment]
I added PHPStanStubLoader and loaded in RectorContainerFactory

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I think it is ready.

**Pull request merged**